tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -254,6 +257,17 @@ bool IsSupportedF8Pattern(
               << " not rewritten into FP8 Custom Call.";
       return false;
     }
+    // One of the operands of select must be zero for the op to be commutative
+    // with dequantization.
+    if (Match(subgraph[i].first, m::Select()) &&
+        !Match(subgraph[i].first->operand(subgraph[i].second == 2 ? 1 : 2),
+               m::Broadcast(m::ConstantScalar(0)))) {
```

Comment:
Can you add a test for the case where the other operand is nonzero?

Reply:
We generally don't have negative tests. Do we need one here?

Reply:
I think it's worthwhile to have one in this case, since otherwise we wouldn't catch bugs like if you hypothetically matched any ConstantScalar instead of a ConstantScalar of 1.

But for the sake of expediency in getting FP8 models working, I'll approve and merge now. Can you add a test in a follow up PR? If you are too busy I can also add one.